ssl/quic/quic_impl.c

```diff
@@ -165,6 +202,7 @@ void ossl_quic_free(SSL *s)
     /* Note: SSL_free calls OPENSSL_free(qc) for us */
 
     SSL_free(qc->tls);
+    ossl_crypto_mutex_free(&qc->mutex);
```

Comment:
Where is the qc->mutex unlocked? Or doesn't it have to be? It looks weird so at least some comment would be appropriate in that case.

Reply:
This is correct, the lock is freed in the locked state.